### PR TITLE
{Util} Add start_powershell_process

### PR DESCRIPTION
**Pull request by fengzhou-msft**
**Description**  

Add `start_powershell_process()` method to include necessary options such as `-NoProfile` when invoking Powershell Command.

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/util.py [hunk 1]
```diff
@@ -1149,3 +1149,8 @@ def handle_version_update():
             refresh_known_clouds()
     except Exception as ex:  # pylint: disable=broad-except
         logger.warning(ex)
+
+
+def start_powershell_process(ps_process_cmd):
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/util.py:1154]: How about `powershell_start_process`, as it is not really a PowerShell process, but starting a process using PowerShell?

```suggestion
def powershell_start_process(ps_process_cmd):
```

File: src/azure-cli/azure/cli/command_modules/util/custom.py [hunk 2]
```diff
@@ -125,8 +125,9 @@ def upgrade_version(cmd, update_all=None, yes=None):  # pylint: disable=too-many
             logger.warning("Exit the container to pull latest image with 'docker pull mcr.microsoft.com/azure-cli' "
                            "or run 'pip install --upgrade azure-cli' in this container")
         elif installer == 'MSI':
+            from azure.cli.core.util import start_powershell_process
             logger.debug("Update azure cli with MSI from https://aka.ms/installazurecliwindows")
-            exit_code = subprocess.call(['powershell.exe', '-No-Profile', "Start-Process msiexec.exe -Wait -ArgumentList '/i https://aka.ms/installazurecliwindows'"])  # pylint: disable=line-too-long
+            exit_code = start_powershell_process("msiexec.exe -Wait -ArgumentList '/i https://aka.ms/installazurecliwindows'")  # pylint: disable=line-too-long
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/util/custom.py:130]: [`-ArgumentList`](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/start-process?view=powershell-7#parameters) is a parameter of `Start-Process`. Should we break it?

File: src/azure-cli-core/azure/cli/core/util.py [hunk 3]
```diff
@@ -1149,3 +1149,8 @@ def handle_version_update():
...
+    import subprocess
+    return subprocess.call(['powershell.exe', '-Command', 'Start-Process {}'.format(ps_process_cmd)])
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/util.py:1156]: Let's add `-NoProfile`.

**Comment by yungezz** [comment]
hi @fengzhou-msft what's the plan of this PR?

**Comment by fengzhou-msft** [comment]
> hi @fengzhou-msft what's the plan of this PR?

Need to further discuss on Pyinstaller impact with subprocess.

**Comment by yungezz** [comment]
> > hi @fengzhou-msft what's the plan of this PR?
> 
> Need to further discuss on Pyinstaller impact with subprocess.

pyinstaller is still on beta without concrete release plan. we can either merge the PR now or close it?

**Comment by jiasli** [comment]
Close as there isn't too many invocation of `powershell.exe`.

**Pull request closed without merging**